server/src/main/java/org/elasticsearch/action/datastreams/lifecycle/GetDataStreamLifecycleAction.java

```diff
@@ -253,6 +253,16 @@ public void writeTo(StreamOutput out) throws IOException {
         public Iterator<ToXContent> toXContentChunked(ToXContent.Params outerParams) {
             return Iterators.concat(Iterators.single((builder, params) -> {
                 builder.startObject();
+                if (globalRetention != null) {
```

Comment:
Do we not want to show empty global configuration if none is configured? I'm trying to think of something analogous in an API we have (to see if it does this, or shows empty) but haven't thought of anything yet.

Reply:
I have been going back and forth on this too. In the same when we have no data streams we show an empty list. `"data_streams": []`. But because this is an object and not a list, I wasn't sure how to approach it.

Reply:
Do you think that having `global_retention: {}` is more descriptive? Like showing that, yes we do support it and it's non set?

Reply:
I changed the implementation to reflect this options

Reply:
I'm also on the fence. I have no strong feeling either way. But maybe it's nice having it always present -- that way you know it's not set, rather than that you're just not looking for it in the right place.

Reply:
Maybe @dakrone has an opinion or knows of some relevant precedent.

Reply:
One argument against this is that now we need to update every yaml rest test that returns a data stream to require the `data_stream_global_retention`, don't we? Otherwise backwards compatibility test runs will fail.

Reply:
Yes, we do not have that many though. I will double check to ensure they are correct.

Reply:
While checking on the global retention in the existing get lifecycle tests, I noticed..... they were not even running :(. I fixed that in https://github.com/elastic/elasticsearch/pull/112210/commits/15fa8e108547027d2c4cf13b05016e204c133b91.

I also decided to merge the two files into one. It seems to me that since the global retention is part of the response it should be part of the basic tests.